src/frog-base.tsx

```diff
@@ -308,6 +309,25 @@ export class FrogBase<
     const imagePaths = getImagePaths(parseHonoPath(path))
     for (const imagePath of imagePaths) {
       this.hono.get(imagePath, async (c) => {
+        const url = getRequestUrl(c.req)
+
+        const query = c.req.query()
+        if (!query.image) {
+          // If the query is doesn't have an image, it is an initial request to a frame.
+          // Therefore we need to get the link to fetch the original image and jump once again in this method to resolve the options,
+          // but now with query params set.
+          const metadata = await getFrameMetadata(url.href.slice(0, -6)) // Stripping `/image` (6 characters) from the end of the url.
+          const frogImage = metadata.find(
+            ({ property }) => property === 'frog:image',
+          )
+          if (!frogImage)
+            throw new Error(
+              'Unexpected error: frog:image meta tag is not present in the frame.',
+            )
+          // Redirect to this route but now with search params and return the response
+          return c.redirect(frogImage.content)
+        }
```

Comment:
I don't understand – isn't `image` as a query parameter always passed (even on the initial request)?

Reply:
Not anymore, now the initial frame image is served at `/image` route without any query params.

Yet in order to render an image, we need them.
For this purpose I introduced `frog:image` meta tag that holds the actual initial frame image url.

Reply:
If it seems redundant - the reason for this is that WC expects a frame image to be updated at the same url, but having an updated image changes the url as it is compressed and stored in query param.

Therefore this trick allows WC to refetch a new image in the same url.

Reply:
See https://events.xyz/api/v2/frame/ed462ce0. (the patch from this pr is applied here)

![Screenshot 2024-04-03 at 23 38 29](https://github.com/wevm/frog/assets/35642018/86a8e6f0-43f8-4187-b378-5b3ac2eb0998)

`frog:image` has the full image and is stored as a meta tag on the frame route level.

`fc:frame:image` doesn't have query params.

![Screenshot 2024-04-03 at 23 39 48](https://github.com/wevm/frog/assets/35642018/613401b7-68cc-47a3-880a-4e8872873edf)

When you visit the `/image` link without query params (specifically, if `image` is not found, but other keys might also be checked), it is redirecting you to a `/image` link **with** query params.